src/main/scala/cromwell/binding/Scope.scala

```diff
@@ -183,4 +174,37 @@ trait Scope {
    */
   def collectAllScatters = Scope.collectAllScatters(Seq(this), Nil)
 
-}
+  /*
+   * Calls and scatters are accessed frequently so this avoids traversing the whole children tree every time.
+   * Lazy because children are not provided at instantiation but rather later during tree building process.
+   * This prevents evaluation from being done before children have been set.
+   *
+   * FIXME: In a world where Scope wasn't monolithic, these would be moved around
+   */
+  lazy val calls: Seq[Call] = collectAllCalls
+  lazy val scatters: Seq[Scatter] = collectAllScatters
+
+  /**
+   *   Recurses up the scope tree until it hits one w/o a parent.
+   *
+   *   FIXME: In a world where Scope wasn't monolithic this would traverse until it hit a RootScope or whatever
+   */
+  @tailrec
+  final def rootScope: Scope = this.parent match {
+    case Some(p) => p.rootScope
+    case None => this
+  }
+
+  // FIXME: In a world where Scope wasn't monolithic, these would be moved out of here
+  def prerequisiteScopes: Set[Scope]
+  def prerequisiteCallNames: Set[FullyQualifiedName]
+
+  /**
+   *  Returns a set of Calls corresponding to the prerequisiteCallNames
+   *
+   *  Dropping any unfound Calls to the floor but we're already validating that all calls are sane at ingest.
+   *  It's icky because it relies on that validation not changing, but ...
+   */
+  lazy val prerequisiteCalls: Set[Scope] = prerequisiteCallNames flatMap rootScope.callByName
+  def callByName(callName: FullyQualifiedName): Option[Call] = calls find { _.name == callName }
```

Comment:
If `callName` parameter is really meant to be a `FullyQualifiedName`, I think it should be `calls find { _.fullyQualifiedName == callName }` ?